### PR TITLE
NetPlay: Refactor some functions into a common header

### PR DESCRIPTION
**Pull request by Techjar**
This moves some functions to a common place so that both sides can use them, which may be needed for future features. Also reduces the amount of code in the client and server a little bit. I also removed an unused struct, and renamed/restructured some things.

### CODE REVIEW

File: Source/Core/Core/NetPlayCommon.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <array>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+namespace NetPlay
+{
+constexpr u32 LZO_IN_LEN = 1024 * 64;
```
Comment (leoetlino) [inline Source/Core/Core/NetPlayCommon.h:18]: Does this actually need to be exposed publicly?

Comment (Techjar) [reply Source/Core/Core/NetPlayCommon.h:18]: It doesn't now I suppose.

File: Source/Core/Core/NetPlayCommon.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+#include <lzo/lzo1x.h>
+
+#include "Common/FileUtil.h"
+#include "Common/IOFile.h"
+#include "Common/MsgHandler.h"
+#include "Common/SFMLHelper.h"
+
+namespace NetPlay
+{
+static constexpr u32 LZO_IN_LEN = 1024 * 64;
```
Comment (leoetlino) [inline Source/Core/Core/NetPlayCommon.cpp:16]: constexpr variables at namespace scope already have internal linkage by default, so `static` is not required here.

File: Source/Core/Core/NetPlayCommon.cpp [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+static constexpr u32 LZO_OUT_LEN = LZO_IN_LEN + (LZO_IN_LEN / 16) + 64 + 3;
```
Comment (leoetlino) [inline Source/Core/Core/NetPlayCommon.cpp:17]: nor here

**Review by leoetlino — APPROVED** [review]

**Pull request merged**